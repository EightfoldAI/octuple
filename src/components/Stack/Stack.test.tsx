import React from 'react';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MatchMediaMock from 'jest-matchmedia-mock';
import { Stack, StackProps } from './';
import { render } from '@testing-library/react';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia: any;

const stackProps: StackProps = {
  fullWidth: false,
  direction: 'horizontal',
  inline: false,
  flexGap: 'm',
  style: {
    backgroundColor: 'aquamarine',
  },
  classNames: 'my-stack-class',
  breakpoints: {},
  children: (
    <>
      <div
        style={{
          height: '200px',
          width: '200px',
          backgroundColor: 'mediumpurple',
        }}
      />
      <div
        style={{
          height: '200px',
          width: '200px',
          backgroundColor: 'mediumpurple',
        }}
      />
      <div
        style={{
          height: '200px',
          width: '200px',
          backgroundColor: 'mediumpurple',
        }}
      />
    </>
  ),
  'data-testid': 'test-stack',
};

describe('Spinner', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  test('Renders without crashing', () => {
    const { container, getByTestId } = render(<Stack {...stackProps} />);
    const stack = getByTestId('test-stack');
    expect(() => container).not.toThrowError();
    expect(stack).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('Stack is vertical', () => {
    const { container, getByTestId } = render(
      <Stack {...stackProps} direction="vertical" />
    );
    const stack = getByTestId('test-stack');
    expect(() => container).not.toThrowError();
    expect(stack.classList.contains('vertical')).toBe(true);
    expect(container).toMatchSnapshot();
  });
});
