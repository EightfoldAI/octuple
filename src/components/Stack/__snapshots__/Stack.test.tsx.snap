// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Renders without crashing 1`] = `
<div>
  <div
    class="stack my-stack-class horizontal undefined gap-m"
    data-testid="test-stack"
    style="background-color: aquamarine;"
  >
    <div
      style="height: 200px; width: 200px; background-color: mediumpurple;"
    />
    <div
      style="height: 200px; width: 200px; background-color: mediumpurple;"
    />
    <div
      style="height: 200px; width: 200px; background-color: mediumpurple;"
    />
  </div>
</div>
`;

exports[`Spinner Stack is vertical 1`] = `
<div>
  <div
    class="stack my-stack-class vertical undefined gap-m"
    data-testid="test-stack"
    style="background-color: aquamarine;"
  >
    <div
      style="height: 200px; width: 200px; background-color: mediumpurple;"
    />
    <div
      style="height: 200px; width: 200px; background-color: mediumpurple;"
    />
    <div
      style="height: 200px; width: 200px; background-color: mediumpurple;"
    />
  </div>
</div>
`;
